q2_diversity/_ordination.py

```diff
@@ -28,3 +30,48 @@ def pcoa(distance_matrix: skbio.DistanceMatrix,
 def pcoa_biplot(pcoa: skbio.OrdinationResults,
                 features: pd.DataFrame) -> skbio.OrdinationResults:
     return skbio.stats.ordination.pcoa_biplot(pcoa, features)
+
+def tsne(distance_matrix: skbio.DistanceMatrix, n_components=3, *, perplexity=30.0, 
+         early_exaggeration=12.0,learning_rate=200.0, n_iter=1000,
+         n_iter_without_progress=300,min_grad_norm=1e-07, metric='euclidean',
+         init='random', verbose=0, random_state=None, method='barnes_hut',
+         angle=0.5, n_jobs=None)-> skbio.OrdinationResults:
+    
+    data = distance_matrix.data
+    ids = distance_matrix.ids
+    long_method_name = "t-distributed stochastic neighbor embedding"
+    tsneData = TSNE(n_components,perplexity,early_exaggeration,learning_rate,
+                    n_iter,n_iter_without_progress,min_grad_norm,metric,init,
+                    verbose,random_state,method,angle,n_jobs).fit_transform(data)
+    
+    axis_labels = ["TSNE%d" % i for i in range(1, n_components + 1)]
+    eigenvalues = [0 for i in axis_labels]
+    
+    return skbio.OrdinationResults(
+        short_method_name="T-SNE",
+        long_method_name=long_method_name,
+        
+        eigvals=pd.Series(eigenvalues, index=axis_labels),
+        samples=pd.DataFrame(tsneData, index=ids,
+                             columns=axis_labels))
+        
+def uMAP(distance_matrix: skbio.DistanceMatrix,n_components=3) -> skbio.OrdinationResults:
+    
+    reducer = umap.UMAP(n_components=n_components)
+   
+    data = distance_matrix.data
+    ids = distance_matrix.ids
+    umap_data = reducer.fit_transform(data)
+    axis_labels = ["UMAP%d" % i for i in range(1, n_components + 1)]
+    eigenvalues = [0 for i in axis_labels]
+    long_method_name = "Uniform Manifold Approximation and Projection"
+
+    return skbio.OrdinationResults(
+        short_method_name="UMAP",
+        long_method_name=long_method_name,
```

Comment:
why is it necessary to use a variable in line 67 for the name?

Reply:
It's not I just tried to stay uniform with the existing code. I can chane that.